Remove XivCommon from the project

Infiziert90 opened the pull request:
- Print description on join is a vanilla function
- Refresh PF listings got extracted from XivCommon and put into its own class

Infiziert90 committed:
Remove XivCommon from the project

keysmashes commented on BetterPartyFinder/Configuration.cs:
```diff
@@ -15,18 +15,17 @@ public class Configuration : IPluginConfiguration {
 
         public bool ShowWhenPfOpen { get; set; }
         public WindowSide WindowSide { get; set; } = WindowSide.Left;
-        public bool ShowDescriptionOnJoin { get; set; } = true;
```
do you know if this will:

- opportunistically remove this value from config whenever config is loaded?
- remove this value from config the next time config is changed?
- leave the value in config indefinitely?
- something else?

keysmashes reviewed (commented):
thank you so much! two questions before I merge

keysmashes commented on BetterPartyFinder/HookManager.cs:
```diff
@@ -0,0 +1,31 @@
+// Code taken from: https://git.anna.lgbt/anna/XivCommon/src/branch/main/XivCommon/Functions/PartyFinder.cs
+public unsafe class HookManager
+{
+    [Signature("48 89 5C 24 ?? 48 89 74 24 ?? 57 48 83 EC 40 0F 10 81")]
```
...github ate my comment...

is this signature likely to make its way into CS?

Infiziert90 committed:
- Adjust bracket style
- Switch to dalamud's window system
- Use ImRaii
- Split-up files as seen necessary
